addons/udes_stock_refactoring/models/stock_location.py

```diff
@@ -0,0 +1,40 @@
+"""UDES Refactoring extensions for the StockLocation model."""
+
+from odoo import fields, models
+
+
+class StockLocation(models.Model):
+    """Stock Location model."""
+
+    _inherit = "stock.location"
+
+    u_is_picking_zone = fields.Boolean(
+        string="Is A Picking Zone",
+        help="Picking Zones are the level to which warehouse-wide Picks are broken down.",
+    )
+    # TODO do we need an ondelete here?
+    u_picking_zone_id = fields.Many2one(
+        "stock.location",
+        string="Picking Zone",
+        help="This location's picking zone, if it is in a picking zone.",
+        compute="_compute_picking_zone_id",
+        readonly=True,
+        store=False,
+    )
+
+    def _compute_picking_zone_id(self):
+        """Set the location that is the picking zone for the location in self.
+
+        A picking zone is a location with u_is_picking_zone == True.
+        If a record is a picking zone, its is set to itself, otherwise successive parent
+        locations are checked, and the first one which is a picking zone is set.
+        If no picking zone is found, False is set.
+        """
+        for record in self:
+            record.u_picking_zone_id = False
+            location = record
+            while location and not record.u_picking_zone_id:
```

Comment:
I'm not sure if it's faster, or exactly the same behaviour, but maybe instead of a while loop, we could do a `Location.search_read([("u_is_picking_zone", "=", True), ("id", "parent_of", record.id)], ["id"])`
Wouldn't work if we need to be able to set multiple picking zone ids within a hierarchy though

Reply:
Implemented in 0b5369ae